instance/multi.go

```diff
@@ -43,11 +43,16 @@ func (m *MultitenantInstancer) Get(instanceID flux.InstanceID) (*Instance, error
 	if err != nil {
 		return nil, errors.Wrap(err, "decoding registry credentials")
 	}
-	regClient := registry.NewClient(
-		creds,
-		log.NewContext(instanceLogger).With("component", "registry"),
-		m.RegistryMetrics.WithInstanceID(instanceID),
-	)
+	var reg registry.Registry
+	{
+		reg = registry.NewRegistry(
+			registry.NewRemoteClientFactory(creds),
```

Comment:
Right. I was more objecting to the naming. (And if there is any way we can avoid the factory altogether)